navigation-material-3/src/main/java/com/google/accompanist/navigation/material3/BottomSheetNavigator.kt

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2023 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.accompanist.navigation.material3
+
+import androidx.compose.foundation.layout.ColumnScope
+import androidx.compose.runtime.Composable
+import androidx.navigation.FloatingWindow
+import androidx.navigation.NavBackStackEntry
+import androidx.navigation.NavDestination
+import androidx.navigation.NavOptions
+import androidx.navigation.Navigator
+
+@ExperimentalMaterial3NavigationApi
+@Navigator.Name("M3BottomSheetNavigator")
+class BottomSheetNavigator: Navigator<BottomSheetNavigator.Destination>() {
+
+    /**
+     * Get the back stack from the [state].
+     */
+    internal val backStack get() = state.backStack
+
+    /**
+     * Dismiss the sheet destination associated with the given [backStackEntry].
+     */
+    internal fun dismiss(backStackEntry: NavBackStackEntry) {
+        state.popWithTransition(backStackEntry, false)
+    }
+
+    override fun navigate(
+        entries: List<NavBackStackEntry>,
+        navOptions: NavOptions?,
+        navigatorExtras: Extras?
+    ) {
+        entries.forEach { entry ->
+            state.push(entry)
```

Comment:
Can we use `pushWithTransition` here and avoid moving the Lifecycle to `RESUMED` until after the bottom sheet is expanded? This would require that you call `onTransitionComplete` when the bottom sheet state settles.